src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/illegalidentifiername/Example2.java

```diff
@@ -0,0 +1,42 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="IllegalIdentifierName">
+      <property name="format"
+        value="(?i)^(?!(record|yield|var|permits|sealed|open|transitive|_)$).+$"/>
+    </module>
+  </module>
+</module>
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.illegalidentifiername;
+
+// xdoc section -- start
+public class Example2 {
+  Integer var = 4; // violation, 'Name 'var' must match pattern'
+  int record = 15; // violation, 'Name 'record' must match pattern'
+  String yield = "yield";
+  // violation above, 'Name 'yield' must match pattern'
+
+  record Record(Record r){} // violation, 'Name 'Record' must match pattern'
+
+  record R(Record record){} // violation, 'Name 'record' must match pattern'
+
+  String yieldString = "yieldString";
+  // ok above, 'yield' is part of broader term, it is not alone
+  record MyRecord(){}
+  // ok above, 'Record' is part of broader term, it is not alone
+  Integer variable = 2;
+  // ok above, 'var' is part of broader term, it is not alone
+
+  int open = 4; // violation, 'Name 'open' must match pattern'
+  Object transitive = "transitive";
```

Comment:
Please review one more time issue description, code must be the same by line numbers.
If this line is on 34th, it must be on 34th in all examples.

Reply:
Done.
I thought it would be fine if the examples differed only in the configuration comments' line numbers. I apologize for any confusion.

Reply:
@chrccl , can you suggest us update for rules wording in issue to make ti clear for next contributors ?

this mistake happens again and again and we would rather minimize time contibutors/maintainers spent on such PRs.

Reply:
@romani ,
The config comments in the examples differ because of certain property name-value tags. 

If you require the Diff in line numbers to be avoided, we could simply add the rule below to the set of rules that we previously updated.
"
_RULES :_

-Line stating `// xdoc section -- start` should be on the same _line number_ for all example files. Blank lines maybe added above to compensate.
"

Reply:
@chrccl , do you think that such extra rule ^^^ woulld make it avoid this "problem" ?

Reply:
@romani , are you referring to me (@MANISH-K-07 ) ?
I'm confused because this is not my PR but was helping from the start.

> do you think that such extra rule ^^^ woulld make it avoid this "problem" ?

We already have good rules which  describe requirement of same Java code. 
Adding this rule that I mentioned above might just eliminate the “_problem_” with line numbers also.

Reply:
@romani, I agree with the proposed rule, but I think it would be beneficial to extend it to include "// xdoc section -- end" as well. Ensuring both start and end comments are on the same line for all example files will provide a more comprehensive solution. In addition, the following rule (already present in the Issue description): 

_'If a code line is present in one example on Line X, it should be located on the same line in all other examples'_ 

along with the new rule to be added, makes the methodology of writing Java code in the examples crystal clear!
So, to conclude, a new possible rule to make the description of the issue more comprehensive and unambiguous could be:

_The line stating // xdoc section -- start and // xdoc section -- end should be on the same line number for all example files. Blank lines may be added below the config comment to compensate._

Reply:
“The line stating // xdoc section -- start and // xdoc section -- end should be on the same line number for all example files. Blank lines may be added below the config comment to compensate.”

This sounds good. It should be enough to guide the future contributors....
:)

Reply:
@MANISH-K-07 ,
It seems like a reasonable rule... let's see what Mr. @romani  thinks about it ;)